### PR TITLE
Create the scaffolding for all commands

### LINKED ISSUE #24: Create the scaffolding for all commands
We should create the scaffolding for all the commands and subcommands so non-devs can provide feedback on the UX

### PR DESCRIPTION
**Pull request by jrcast**
1. Create the scaffolding for all commands
    1. Add `forgeops status` command
    1. Add `forgeops apply base` command
    1. Add `forgeops apply directory` command
    1. Add `forgeops apply apps` command
    1. Add `forgeops delete base` command
    1. Add `forgeops delete directory` command
    1. Add `forgeops delete apps` command
1. Cleaned up linter errors

Closes #24

### CODE REVIEW

File: pkg/health/ctrl.go [hunk 1]
```diff
@@ -6,10 +6,14 @@ import (
 	"github.com/ForgeRock/forgeops-cli/internal/factory"
 	"github.com/ForgeRock/forgeops-cli/internal/k8s"
 	"github.com/ForgeRock/forgeops-cli/internal/printer"
+	"github.com/pkg/errors"
 )
 
-// HealthFromBytes deserialize from bytes
-func HealthFromBytes(hbytes []byte) (*Health, error) {
+// ErrNotAllHealthy only boolean expressions are permitted
+var ErrNotAllHealthy error = errors.New("not all checks were successul")
```
Comment (lee-baines) [inline pkg/health/ctrl.go:13]: s/successul/successful/

**Review by lee-baines — APPROVED** [review]

**Comment by jrcast** [comment]
@dgoldssfo @shankar-forgerock Take a look at the changes under `/cmd/`. The files under `/docs` are autogenerated so keep that in mind. The PR description lists the new commands added.

File: docs/forgeops_apply_apps.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+```
+  -h, --help   help for apps
+```
+
+### Options inherited from parent commands
+
+```
+      --as string                      Username to impersonate for the operation
+      --as-group stringArray           Group to impersonate for the operation, this flag can be repeated to specify multiple groups.
```
Comment (shankar-forgerock) [inline docs/forgeops_apply_apps.md:37]: May be a non-issue, but in any case: If multiple groups are assigned, will this use the "union" (most permissible) or "intersection" (least permissible) permission?

File: docs/forgeops_apply_apps.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      --cache-dir string               Default cache directory (default "/home/jcastillo/.kube/cache")
+      --certificate-authority string   Path to a cert file for the certificate authority
+      --client-certificate string      Path to a client certificate file for TLS
+      --client-key string              Path to a client key file for TLS
+      --cluster string                 The name of the kubeconfig cluster to use
+      --context string                 The name of the kubeconfig context to use
+      --insecure-skip-tls-verify       If true, the server's certificate will not be checked for validity. This will make your HTTPS connections insecure
```
Comment (shankar-forgerock) [inline docs/forgeops_apply_apps.md:44]: I guess by default certificate is verified. 

Should we call it 'skip-tls-cert-verify'?

**Review by shankar-forgerock — APPROVED** [review]

File: docs/forgeops_apply_apps.md (hunk 2 above)
Comment (jrcast) [reply docs/forgeops_apply_apps.md:37]: these flags are inherited from a k8s library.

File: docs/forgeops_apply_apps.md (hunk 3 above)
Comment (jrcast) [reply docs/forgeops_apply_apps.md:44]: these flags are inherited from a k8s library.

File: cmd/apply.go [hunk 4]
```diff
@@ -83,14 +84,80 @@ var dsOperator = &cobra.Command{
...
+var forgeopsBase = &cobra.Command{
+	Use:     "base",
+	Aliases: []string{"fb"},
+	Short:   "Deploy the ForgeRock base resources",
+	Long: `
+    Deploy the base resources of the ForgeRock cloud deployment:
+    * Apply the base resources of ForgeRock cloud deployment
+    * use --tag to specify a different version to deploy`,
+	Example: `
+      # Deploy the base resources listed in the "latest" release of Forgeops.
```
Comment (dgoldssfo) [inline cmd/apply.go:96]: Maybe s/Forgeops/the Forgeops repository/

File: cmd/apply.go [hunk 5]
```diff
@@ -83,14 +84,80 @@ var dsOperator = &cobra.Command{
...
+      forgeops apply base
+
+      # Deploy the base resources listed in a specific release of Forgeops.
```
Comment (dgoldssfo) [inline cmd/apply.go:99]: Maybe s/Forgeops/the Forgeops repository/ (throughout the code)

File: cmd/apply.go [hunk 6]
```diff
@@ -83,14 +84,80 @@ var dsOperator = &cobra.Command{
...
 }
 
+var forgeopsBase = &cobra.Command{
+	Use:     "base",
+	Aliases: []string{"fb"},
+	Short:   "Deploy the ForgeRock base resources",
+	Long: `
+    Deploy the base resources of the ForgeRock cloud deployment:
+    * Apply the base resources of ForgeRock cloud deployment
+    * use --tag to specify a different version to deploy`,
```
Comment (dgoldssfo) [inline cmd/apply.go:94]: Any reason why "use" and not "Use"? (if you change, change throughout the code)

File: cmd/apply.go [hunk 7]
```diff
@@ -83,14 +84,80 @@ var dsOperator = &cobra.Command{
...
+	Short:   "Deploy the ForgeRock DS resources",
+	Long: `
+    Deploy the directory service resources of the ForgeRock cloud deployment:
+    * Apply the directory service resources of ForgeRock cloud deployment
+    * use --tag to specify a different version to deploy`,
+	Example: `
+      # Deploy the directory service resources listed in the "latest" release of Forgeops.
+      forgeops apply directory
+
+      # Deploy the directory service resources listed in a specific release of Forgeops.
```
Comment (dgoldssfo) [inline cmd/apply.go:121]: ...of the Forgeops repository

File: cmd/apply.go [hunk 8]
```diff
@@ -83,14 +84,80 @@ var dsOperator = &cobra.Command{
...
+	DisableAutoGenTag: true,
+}
+
+var forgeopsApps = &cobra.Command{
+	Use:     "apps",
+	Aliases: []string{"fa"},
+	Short:   "Deploy the ForgeRock apps (AM, IDM, UI)",
+	Long: `
+    Deploy the ForgeRock apps (AM, IDM, UI):
+    * Apply the ForgeRock apps
```
Comment (dgoldssfo) [inline cmd/apply.go:137]: Deploy or Apply?

File: cmd/delete.go [hunk 9]
```diff
@@ -68,6 +69,68 @@ var deleteDsOperator = &cobra.Command{
 	DisableAutoGenTag: true,
 }
 
+var deleteForgeopsBase = &cobra.Command{
+	Use:     "base",
+	Aliases: []string{"fb"},
+	Short:   "Remove the ForgeRock base resources",
+	Long: `
+    Remove the base resources of the ForgeRock cloud deployment:
```
Comment (dgoldssfo) [inline cmd/delete.go:77]: Pick one (remove or delete) and stick with it (throughout)

**Review by dgoldssfo — APPROVED** [review]
I'm not sure whether I got everything, but after a certain point, stuff looked generated.

**Pull request merged**